pkg/reconcilermanager/controllers/garbage_collector.go

```diff
@@ -171,31 +170,71 @@ func (r *RepoSyncReconciler) deleteRoleBinding(ctx context.Context, reconcilerRe
 	return nil
 }
 
-func (r *RepoSyncReconciler) deleteHelmValuesFileRefCopies(ctx context.Context, rs *v1beta1.RepoSync) error {
-	if rs == nil || rs.Spec.SourceType != string(v1beta1.HelmSource) || rs.Spec.Helm == nil || len(rs.Spec.Helm.ValuesFileRefs) == 0 {
-		return nil
+// deleteUnusedHelmConfigMapCopies deletes helm values file ConfigMap copies
+// that are no longer required.
+func (r *RepoSyncReconciler) deleteUnusedHelmConfigMapCopies(ctx context.Context, rs *v1beta1.RepoSync) error {
+	rsRef := client.ObjectKeyFromObject(rs)
+	var cmNamesToKeep map[string]struct{}
+	if rs.Spec.SourceType == string(v1beta1.HelmSource) && rs.Spec.Helm != nil {
+		cmNamesToKeep := make(map[string]struct{}, len(rs.Spec.Helm.ValuesFileRefs))
```

Comment:
I deliberately extracted getHelmConfigMapCopyRef so that we don't have to pass the computed []ValuesFileRefs through multiple layers of abstraction, so we can have cleaner, more consistent code. Yes, it re-hashes the names, but it's not that expensive.

Reply:
I guess I could in-line the deleteUnusedHelmConfigMapCopies code into upsertHelmConfigMaps to avoid hashing them twice. I just figured this was easier to read.

Reply:
Done.
This also fixes the garbage collection when SourceType changed or ValuesFileRefs was removed.

This makes me think... are other resources being garbage collected when the config changes? Like when someone switches from one Secret to another? Or are the copies being left behind?

Reply:
upsertServiceAccount and upsertRoleBinding probably don't need to garbage collect, because they're always required and don't change name based on user input.